pkg/util/slowset.go

```diff
@@ -25,68 +25,86 @@ const (
 	slowSetSleepDelay = 100 * time.Millisecond
 )
 
-type SlowSet struct {
+type FailureTimestamp[T ~string] struct {
+	FailureType T
+	Timestamp   time.Time
+}
+
+type SlowSet[T ~string] struct {
```

Comment:
```
// SlowSet is a set of API objects that should be synced at slower rate. Key is typically the object
// namespace + name. 
// Reason why is the object in the queue is recorded in type `T`. For example,
// `SlowSet[v1.ClaimResourceStatus]` will remember if the object is in the queue because it status is
// `PersistentVolumeClaimControllerResizeInfeasible` or `PersistentVolumeClaimNodeResizeInfeasible`.
// `Remove()` will then be able to remove the object from the slow set when ControllerResize becomes
// feasible or NodeResize becomes feasible.
```